quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs

```diff
@@ -40,27 +43,81 @@ impl Default for SingleFileMetastoreFactory {
     }
 }
 
+fn extract_polling_period_from_uri(uri: &str) -> (String, Option<Duration>) {
```

Comment:
I'm not fond of this. Unfortunately, this adds another way to configure quickwit, and we can't detect issues at config parsing time unless we add some additional validation logic.

I hope in the future we can get rid of this and resolve `Metastore` and `Storage` with an optional config object:
`Metastore::resolve("s3://...", Some(MetastoreConfig { ... }))`.

Reply:
Yes it is not great. Right now if the users gets the DSL wrong we don't spot anything & don't log anything.